app/models/ability.rb

```diff
@@ -12,6 +12,14 @@ def initialize(user)
     can :crud, User, id: user.id
     can :crud, User if user.admin?
 
+    # visibility of email address in user profile
+    can :read_email, User, id: user.id if !user.hide_email?
+    can :read_email, User if user.admin?
+    can :read_email, User do |other_user|
+      (user.confirmed? && supervises?(other_user, user)) ||
```

Comment:
Since we always require the user to be confirmed, how about:

```ruby
can :read_email, User do |other_user|
  user.confirmed? && (supervises?(other_user, user)) || !other_user.hide_email?)
end
```